src/AppBundle/Controller/UserController.php

```diff
@@ -0,0 +1,354 @@
+        $form2 = $this->get('form.factory')->create(new UserType(), $usr);
+
+        // START
+        // BUILD FORM TO EDIT ALL USERS INFO //
+        // RENDER VARIABLES AND PAGE //
+        return $this->render('@App/App/adminProfile.html.twig', array(
+            'id' => $id,
+            'usr' => $usr,
+            'usrs' => $usrs,
+            'dsh' => $dsh,
+            'dshs' => $dshs,
+            'paginUsrs' => $paginUsrs,
+            'paginDashs' => $paginDshs,
+            'form' => $form->createView(),
+//            'form4' => $form4->createView(),
+            'form2' => $form2->createView(),
+            'newUser' => $newUser
+        ));
+    }
+
+
+    public function deleteDashboardAction($usrid, $id)
+    {
+        $em = $this->getDoctrine()->getManager();
+        $usr = $this->container->get('security.context')->getToken()->getUser();
+        $usrid = $usr->getId();
+        $dashboard = $em->getRepository('AppBundle:Dashboard')->find($id);
+        if ($dashboard != null) {
+            $em->remove($dashboard);
+            $em->flush();
+        }
+        return $this->redirectToRoute('app_admin_profile', array(
+            'id' => $id,
+            'dashboard' => $dashboard,
+            'usrid' => $usrid
+        ));
+    }
+
+    // START EDIT OWN USER INFO //
+    public function editUserAction(Request $request, $id)
+    {
+        $em = $this->getDoctrine()->getManager();
+        $usr = $em->getRepository('AppBundle:User')->find($id);
+        $usrs = $em->getRepository('AppBundle:User')->findAll();
+        $em->flush();
+
+        // START EDIT USER //
+        $form2 = $this->get('form.factory')->create(new UserType(), $usr);
+
+        if ($form2->handleRequest($request)->isValid()) {
+
+            // $file stores the uploaded image file
+            /** @var Symfony\Component\HttpFoundation\File\UploadedFile $file */
+            $file = $usr->getImage();
+
+            // Generate a unique name for the file before saving it
+            $fileName = md5(uniqid()) . '.' . $file->guessExtension();
+
+            // Move the file to the directory where images are stored
+            $file->move(
+                $this->getParameter('img_directory'),
+                $fileName
+            );
+
+            // Update the 'image' property to store the img file name
+            // instead of its contents
+            $usr->setImage($fileName);
+
+            $em->merge($usr);
+            $em->flush();
+
+            return $this->redirectToRoute('app_admin_profile', array('id' => $id));
+        }
+        // END FORM EDIT USER //
+
+        return $this->render('@App/App/adminProfile.html.twig', array(
+            'id' => $id,
+            'usr' => $usr,
+            'usrs' => $usrs,
+            'form2' => $form2->createView()
+        ));
+    }
+    // END EDIT OWN USER INFO //
+
+    // START EDIT ALL USERS INFO //
+    public function editUsersAction(Request $request, $id)

```

Comment:
This function is used to edit the information of an individual user other than myself.